plugins/BEdita/API/src/Controller/ObjectsController.php

```diff
@@ -61,12 +90,94 @@ public function index()
      */
     public function view($id)
     {
+        $conditions = ['deleted' => 0];
+        if ($this->objectType) {
+            $conditions['object_type_id'] = $this->objectType->id;
+        }
         $object = $this->Objects->get($id, [
             'contain' => ['ObjectTypes'],
+            'conditions' => $conditions
+        ]);
+
+        $this->set(compact('object'));
+        $this->set('_serialize', ['object']);
+    }
+
+    /**
+     * Add a new object.
+     *
+     * @return void
+     * @throws \Cake\Network\Exception\BadRequestException Throws an exception if submitted data is invalid.
+     */
+    public function add()
+    {
+        $this->request->allowMethod('post');
+
+        $object = $this->Objects->newEntity($this->request->data);
+        $object->type = $this->request->data('type');
```

Comment:
I think that here is required a check on consistency between `data('type')` and the endpoint used. For example I could send
```http
POST /documents

{
    "type": "users",
    "attributes": {
        "title": "New document"
    }
}
```

A `400 Bad Request` should be expected (missing unit test too)